### PR TITLE
Fixes for dev httr2

### LINKED ISSUE #30: Mock file can't be source due to `<` character
Is this occurring on anyone else's side? I have `cache = <environment>` just near the end of my mock file. If I replace it with `rlang::caller_env()`, then the file can be sourced and a test will pass.

End of my mock file:
`cache = <environment>, class = "httr2_response")`

### PR DESCRIPTION
**Pull request by hadley**
Fixes #30

### CODE REVIEW

File: R/zzz.R [hunk 1]
```diff
@@ -0,0 +1,5 @@
+httr2_1.0.0 <- NULL
```
Comment (hadley) [inline R/zzz.R:1]: Possibly a premature optimisation, but `packageVersion()` is reasonable slow, and it seems like `with_mock` gets called in a lot of places.

File: tests/testthat/test-mock-api.R [hunk 2]
```diff
@@ -25,7 +25,7 @@ with_mock_api({
       request("api/NOTAFILE/") %>%
         req_url_query(a = 1) %>%
         req_perform(),
-      "api/NOTAFILE/?a=1"
+      if (httr2_1.0.0) "/api/NOTAFILE/?a=1" else "api/NOTAFILE/?a=1"
```
[the remark this answers is not in the document]
Comment (nealrichardson) [reply tests/testthat/test-mock-api.R:28]: Is this even a valid request? That's not a proper URL. 

There was some long-ago reason I supported this in the early days of httptest, just making arbitrary mock files, but seems hacky and unnecessary today. I'd be ok with this workaround right now but maybe this feature should be removed and these tests modified to use valid URLs.

File: R/zzz.R [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+.onLoad <- function(...) {
+  httr2_1.0.0 <<- utils::packageVersion("httr2") > "0.2.3"
```
Comment (nealrichardson) [inline R/zzz.R:4]:
```suggestion
  # packageVersion() can be slow, so let's only do this check once
  httr2_1.0.0 <<- utils::packageVersion("httr2") > "0.2.3"
```

File: R/expect-request-header.R [hunk 4]
```diff
@@ -82,5 +82,13 @@ expect_request_header <- function(expr,
     }
   }
 
-  httr2::with_mock(header_mocker, expr)
+  with_mock(header_mocker, expr)
+}
+
+with_mock <- function(mock, code) {
+  if (httr2_1.0.0) {
+    httr2::with_mocked_responses(mock, code)
```
Comment (nealrichardson) [inline R/expect-request-header.R:90]: I think this needs to be `getFromNamespace` if I am to submit an update to CRAN before httr2 lands; this currently raises a check warning.

File: tests/testthat/test-mock-api.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (nealrichardson) [reply tests/testthat/test-mock-api.R:28]: https://github.com/nealrichardson/httptest2/blob/main/R/mock-api.R#L64-L66 is where this is currently handled, if we wanted to patch it up further.

[the remark this answers is not in the document]
Comment (hadley) [reply tests/testthat/test-mock-api.R:28]: Hmmm, in that case, and given it's one test in one package, I think it'll be easiest if I just update their test/code.

**Review by nealrichardson — APPROVED** [review]
Thank you!

**Pull request merged**